lti_consumer/lti_1p1/consumer.py

```diff
@@ -288,6 +289,17 @@ def generate_launch_request(self, resource_link_id):
         if self.lti_custom_parameters:
             lti_parameters.update(self.lti_custom_parameters)
 
+        if self.xblock:
+            for processor in self.xblock.get_parameter_processors():
+                try:
+                    default_params = getattr(processor, 'lti_xblock_default_params', {})
+                    lti_parameters.update(default_params)
+                    lti_parameters.update(processor(self.xblock) or {})
+                except Exception:  # pylint: disable=broad-except
+                    # Log the error without causing a 500-error.
+                    # Useful for catching casual runtime errors in the processors.
+                    log.exception('Error in XBlock LTI parameter processor "%s"', processor)
+
```

Comment:
For this specific use case, I'd suggest an approach similar to what I've done for LTI 1.3:
1. Add an extra settings variable to the class: https://github.com/edx/xblock-lti-consumer/blob/master/lti_consumer/lti_1p3/consumer.py#L60 and a public update method for it https://github.com/edx/xblock-lti-consumer/blob/master/lti_consumer/lti_1p3/consumer.py#L448-L454
2. If there's anything on it, update the parameters on launch (same as on https://github.com/edx/xblock-lti-consumer/blob/master/lti_consumer/lti_1p3/consumer.py#L301-L303):
```suggestion
        # Extra claims - from custom parameter processors
        if self.extra_claims:
            lti_message.update(self.extra_claims)
```

Then, on the `xblock` code, when doing an LTI 1.1 launch, add the parameter processors (see https://github.com/edx/xblock-lti-consumer/blob/master/lti_consumer/lti_xblock.py#L1027 for reference):
```
# From XBlock code
for processor in self.get_parameter_processors():
  try:
      default_params = getattr(processor, 'lti_xblock_default_params', {})
      lti_consumer.set_extra_claims(processor(self) or default_params)
  except Exception:  # pylint: disable=broad-except
      # Log the error without causing a 500-error.
      # Useful for catching casual runtime errors in the processors.
      log.exception('Error in XBlock LTI parameter processor "%s"', processor)
```